script.js

```diff
@@ -0,0 +1,200 @@
+let webBrands = [];
+let brands=[];
+let resultbrands=[];
+let productNode;
+const proxyurl = "https://cors-anywhere.herokuapp.com/";
+const url = `https://wellspent-ethical.herokuapp.com/api/brands`;
+
+
+var linksList = document.querySelectorAll("[id^=product-]");
+linksList.forEach(function(product) {
+  const node = product.childNodes[0];
+  const label = node.getAttribute("aria-label");
+  let brand = label.toLowerCase();
+  webBrands.push(brand);
+});
+
+
+ // site that doesn’t send Access-Control-*
+fetch(proxyurl + url) // https://cors-anywhere.herokuapp.com/https://example.com
+  .then(resultbrandsonse => resultbrandsonse.text())
+  .then(contents => {
+    resultbrands = JSON.parse(contents);
+    brands = resultbrands;
+
+
+    const finalResult = brands.result.map((data)=>{
+    const result =   webBrands.filter((webBrand)=>{
+      const splitedBrand= data.BrandName.toLowerCase().split(" ")[0];
+      if(webBrand.indexOf(splitedBrand)==0){
+        return webBrand.includes(data.BrandName.toLowerCase())
+      }
+      })
+      return result.length > 0 ? [data.BrandName] : [];
+    });
+    const filteredBrand = Promise.all(finalResult)
+    .then((res)=>{
+        res.map((brand,index, array) => {
+          if(brand.length > 0){
+          linksList.forEach(function(product) {
+            const node = product.childNodes[0];
+            const label = node.getAttribute("aria-label");
+            let productNodeText = label.toLowerCase();
+            if(productNodeText.includes(brand[0].toLowerCase())){
+                product.style["border-color"] =resultbrands.colourMap[resultbrands.result[index].OverallScore];
+                product.classList.add('product');
```

Comment:
All the code here is one block, having everything inside the fetch method making it hard to understand/track.
You should separate your code more, for example, you could write all the style lines in another function called addStyle or anything related and called it here.